tests/unit/test_solvers/test_algebraic_solver.py

```diff
@@ -295,6 +295,7 @@ def test_solve_with_input(self):
         # Solve
         solver = pybamm.AlgebraicSolver()
         solution = solver.solve(model, np.linspace(0, 1, 10), inputs={"value": 7})
+        model.print_parameter_info()
```

Comment:
I think this line can be deleted

Reply:
Both the property `model.parameters` and the function `model.print_parameter_info()` use caching to avoid having to recalculate the parameters when the function is called several times, see for example https://github.com/pybamm-team/PyBaMM/pull/1471/files#diff-579718d1877c511c7d203b3e4dfbbb093d1fb0b728efeb2069431ee27aa57250R275-R281 (the first time, `self._parameters` is None and then gets updated, the second time it already has been updated and so doesn't change).
Therefore this doesn't keep track of any changes in the model and just returns whatever it was the first time you called it.
This is probably overkill as this function will only be used in model building and model exploration, so doesn't need to be optimized like this. So can you edit the code to remove the caching?

Reply:
Regarding the original comment, it would be clearer if the line
```python
if input_param.domain == []:
    self._parameter_info += f"{input_param.name} (InputParameter)\n"
```
was tested by test_base_model (i.e. add an inputparameter without domain to that test model)

Reply:
I've made the necessary changes, I think everything should work fine now

Reply:
thanks! merging